microcosm_pubsub/chain/statements.py

```diff
@@ -1,37 +1,59 @@
+from microcosm_pubsub.chain import Chain
+
+
 class ArgumentExtractor:
-    def __init__(self, name, key, key_property=None):
-        self.name = name
-        self.key = key
-        self.key_property = key_property
+    """
```

Comment:
Reimplemented to:
 - allow `this` to use dotted syntax: `foo.bar`
 - allow `that` to be defined "later" via the `as_` builder

That is:
```
extract("foo", "bar", "baz")
``
can now be defined as:
```
assign("bar.baz").as_("foo")
```